lib/app_profiler/railtie.rb

```diff
@@ -41,6 +44,9 @@ class Railtie < Rails::Railtie
           app.middleware.insert_before(0, Viewer::SpeedscopeRemoteViewer::Middleware)
         end
         app.middleware.insert_before(0, AppProfiler.middleware)
+        ActiveSupport::ForkTracker.after_fork do
+          AppProfiler::Middleware::UploadAction.start_process_queue_thread
+        end
```

Comment:
This starts the thread for all usage contexts of the profiler, which I think might be incorrect. If the developer never wants to upload asynchronously, this thread has no purpose. Is it possible to do this lazily?

Reply:
This was a good call out, thanks 👍